### DISCUSSION TITLE
Shell Mode Feature

### DISCUSSION THREAD

**Discussion by NoobforAl** [curl/trurl#209] [Ideas]
Hi,
Is possible add shell mode in this project ?
for example I have one Url and I want test many query and check result .
For Example:
> trurl --url  https://example.com -sh

command `-sh` is going to shell mode.
and we can edit url and request.

for example replace the host name:  
```sh
example.com >> set host=test.com
test.com >>
```
or get port and path:
```sh
test.com >> get  {port} 
443
```
and after this change and check request:
```sh
test.com >> sendReq
Not Found.
test.com >>
```
we can use all feature curl and trurl for url.

**Comment by jacobmealey** [comment]
For using multiple queries or hostnames and stuff  `--iterate` might already do most of what you want. I'm not sure I understand the benefits of making an interactive shell - other than holding state between runs which could be done with a bash script or something.

**Reply by NoobforAl** [reply]
That true, I forgot this [**advice**](https://catb.org/~esr/writings/unix-koans/ten-thousand.html) XD.
But I think most be better this ability added ( not this project but can use this project ) because for work with a API or URL( I means is if want test different parameters ) normally we need **Postman** or some like tools. 
but I don't see command line tools for handle this ( or I don't find anything on the internet ).
